modules/airavata-helix/helix-spectator/src/main/java/org/apache/airavata/helix/impl/participant/GlobalParticipant.java

```diff
@@ -55,9 +55,9 @@
         for (String taskClass : taskClasses) {
             TaskFactory taskFac = context -> {
                 try {
-                    return AbstractTask.class.cast(Class.forName(taskClass).newInstance())
-                            .setCallbackContext(context)
-                            .setTaskHelper(new TaskHelperImpl());
+                    AbstractTask task = AbstractTask.class.cast(Class.forName(taskClass).newInstance());
+                    task.setParticipant(GlobalParticipant.this);
+                    return task.setCallbackContext(context).setTaskHelper(new TaskHelperImpl());
```

Comment:
Use AbstractTask.set.set format. Do not introduce new variable  as you are immediately returning it